app/models/audit/event/whoami.rb

```diff
@@ -0,0 +1,78 @@
+module Audit
+  module Event
+    class Whoami
+
+      # NOTE: "role_id" may refer to a user or host.
+      def initialize(
+        client_ip:,
+        role_id:,
+        success:
+      )
+        @client_ip = client_ip
+        @role_id = role_id
+        @success = success
+
+        # Implements `==` for audit events
+        @comparable_evt = ComparableEvent.new(self)
+      end
+
+      # NOTE: We want this class to be responsible for providing `progname`.
+      # At the same time, `progname` is currently always "conjur" and this is
+      # unlikely to change.  Moving `progname` into the constructor now
+      # feels like premature optimization, so we ignore reek here.
+      # :reek:UtilityFunction
+      def progname
+        Event.progname
+      end
+
+      def severity
+        attempted_action.severity
+      end
+
+      def to_s
+        message
+      end
+
+      def message
+        "#{@role_id.role_id} checked identity using WhoAmI"
+      end
+
+      def message_id
+        'identity-check'
+      end
+
+      def structured_data
+        {
+          role_id = @role_id.role_id
```

Comment:
i would move this line to a new method and use memoization
```
def role_id
  @role_id ||= @role.role_id
end
```

Note that i changed your `@role_id` to `@role` as (i think) it better represents the object

Reply:
will do, thanks.